File: C#/0001.TwoSum/TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var map = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            int complement = target - nums[i];
            if (map.ContainsKey(complement)) {
                return new int[] { map[complement], i };
            }
            if (!map.ContainsKey(nums[i])) {
                map.Add(nums[i], i);
            }
        }
        throw new ArgumentException("No two sum solution");
    }
}

File: JavaScript/0001.TwoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    const numbersLookingFor = new Map();

    for (let i = 0; i < nums.length; i++) {
        const lookingFor = target - nums[i];

        if (numbersLookingFor.has(lookingFor)) {
            return [numbersLookingFor.get(lookingFor), i];
        }

        numbersLookingFor.set(nums[i], i);
    }

    return [];
};